cups/dest.c

```diff
@@ -224,7 +224,7 @@ static void		cups_dnssd_unquote(char *dst, const char *src,
 			                   size_t dstsize);
 static int		cups_elapsed(struct timeval *t);
 #endif /* HAVE_DNSSD */
-static int              cups_enum_dests(http_t *http, unsigned flags, int msec, int *cancel, cups_ptype_t type, cups_ptype_t mask, cups_dest_cb_t cb, void *user_data);
+static int              cups_enum_dests(http_t *http, unsigned flags, int msec, const int *cancel, cups_ptype_t type, cups_ptype_t mask, cups_dest_cb_t cb, void *user_data);
```

Comment:
const here could cause optimization errors (with the compiler assuming that "cancel" won't change)

Reply:
Will it change though? It doesn't seem to be

Reply:
@AtariDreams The point of this argument is for the "cancel" integer to be used by another thread to cause the enumeration to stop. Thus, while `cups_enum_dests` won't change the value, another thread might. And compilers regularly make optimization choices for `const` data that are not compatible with multi-threaded code.